remotes/docker/fetcher.go

```diff
@@ -150,6 +153,47 @@ func (r dockerFetcher) Fetch(ctx context.Context, desc ocispec.Descriptor) (io.R
 	})
 }
 
+type HTTPError struct {
```

Comment:
I didn't love to have this error under `docker`. Then I realized we have ErrUnexpectedStatus since #4523! So I'm going to update this PR tomorrow.

Reply:
Updated.